src/google/cloud/ndb/query.py

```diff
@@ -1618,6 +1615,369 @@ def _option(self, name, given, options=None):
 
         return None
 
+    def run_to_queue(self, queue, conn, options=None, dsquery=None):
+        """Run this query, putting entities into the given queue."""
+        raise exceptions.NoLongerImplementedError()
+
+    def iter(
+        self,
+        limit=None,
+        *,
+        keys_only=None,
+        projection=None,
+        offset=None,
+        batch_size=None,
+        prefetch_size=None,
+        produce_cursors=False,
+        start_cursor=None,
+        end_cursor=None,
+        deadline=None,
+        read_policy=None,  #  _datastore_api.EVENTUAL,  # placeholder
+        options=None,
+    ):
+        """Get an iterator over query results.
+
+        Args:
+            keys_only (bool): Return keys instead of entities.
+            projection (list[str]): The fields to return as part of the query
+                results.
+            offset (int): Number of query results to skip.
+            limit (Optional[int]): Maximum number of query results to return.
+                If not specified, there is no limit.
+            batch_size (Optional[int]): Number of results to fetch in a single
+                RPC call. Affects efficiency of queries only. Larger batch
+                sizes use more memory but make fewer RPC calls.
+            prefetch_size (Optional[int]): Overrides batch size for first batch
+                returned.
+            produce_cursors (bool): Whether to generate cursors from query.
+            start_cursor: Starting point for search.
+            end_cursor: Endpoint point for search.
+            deadline (Optional[int]): Override the RPC deadline, in seconds.
+            read_policy: Defaults to `ndb.EVENTUAL` for potentially faster
+                query results without having to wait for Datastore to apply
+                pending changes to all returned records.
+            options (QueryOptions): DEPRECATED: An object containing options
+                values for some of these arguments.
+
+        Returns:
+            QueryIterator: An iterator.
+        """
+        raise NotImplementedError
+
+    __iter__ = iter
+
+    def map(
```

Comment:
This seems a over engineered and confusing. I wonder if anyone's actually using this? Surely people could just use `iter` and map and/or reduce as they see fit.

Reply:
Aside from it being in the docs, I can't find evidence that someone is using or recommending using this (unlike utils, where I did find third-party book examples using them).